src/ScriptExport/Types.hs

```diff
@@ -58,55 +76,130 @@ data ScriptQuery = ScriptQuery
      @since 1.0.0
 -}
 runQuery :: ScriptQuery -> Builders -> Servant.Handler Aeson.Value
-runQuery s =
+runQuery (ScriptQuery name param) =
   maybe
     (Servant.throwError Servant.err404 {Servant.errBody = "Builder not found"})
-    ($ s.paramsPayload)
-    . Map.lookup s.name
+    (toServantErr . runExcept . handleServe param)
+    . Map.lookup name
     . getBuilders
+  where
+    toServantErr (Left err) =
+      Servant.throwError
+        Servant.err400
+          { Servant.errBody = (LBS.pack . unpack) err
+          }
+    toServantErr (Right x) = pure x
+
+{- | Possible data to request.
+
+     @since 2.0.0
+-}
+data ServeElement where
+  ServeRawScriptExport ::
+    forall (a :: Type) (param :: Type).
+    (Aeson.FromJSON param, Aeson.ToJSON a) =>
+    RawScriptExport ->
+    Linker param (ScriptExport a) ->
+    ServeElement
+  ServeJSON ::
+    forall (s :: Type).
+    (Aeson.ToJSON s) =>
+    s ->
+    ServeElement
+  ServeJSONWithParam ::
+    forall (p :: Type) (s :: Type).
+    (Aeson.FromJSON p, Aeson.ToJSON s) =>
+    (p -> s) ->
+    ServeElement
+
+{- | Handle `ServeElement` and returns JSON.
+
+     @since 2.0.0
+-}
+handleServe :: Maybe Aeson.Value -> ServeElement -> Except Text Aeson.Value
+handleServe _ (ServeJSON x) = pure $ Aeson.toJSON x
+handleServe (Just arg) (ServeJSONWithParam f) =
+  case Aeson.fromJSON arg of
+    Aeson.Error e ->
+      throwError $ pack e
+    Aeson.Success v' ->
+      pure . Aeson.toJSON $ f v'
+handleServe (Just arg) (ServeRawScriptExport scr linker) =
+  case Aeson.fromJSON arg of
+    Aeson.Error e ->
+      throwError $ pack e
+    Aeson.Success v' ->
+      case runLinker linker scr v' of
+        Left e -> throwError . pack . show $ e
+        Right x -> pure . Aeson.toJSON $ x
+handleServe Nothing (ServeRawScriptExport scr _) = pure $ Aeson.toJSON scr
+handleServe _ _ = throwError "Incorrect Parameter"
```

Comment:
This is fine, but it would be nice to have a slightly better error message here.
For example, if we have `ServeRawScriptExport`, and `Nothing` as the parameter argument, we can say `"Querying a RawScriptExport, but passed no parameters."`

Reply:
`ServeRawScriptExport` would work without a parameter, it would export raw scripts, but I got the point. I'll add better error for `ServeJSONWithParam` which expects parameter always.